### PR TITLE
[Windows] DXVA: minor code improvements

### PR DESCRIPTION
**Pull request by thexai**
## Description
DXVA: minor improvements

## Motivation and context
All are very simple changes in separated commits. 

~The fist two~ not changes the functionality, only removes redundant variables/parameters.

~The third has a minimal functional change enabling 10 bit intermediate target also when "High precision processing" setting is enabled and  source is 10 bit SDR.~

~It's not used much today, but 10-bit x264 encodings still exist.~

## How has this been tested?
Windows x64

## What is the effect on users?
~May improve video quality when source is 10 bit SDR and "High precision processing" is enabled (default) and "Use 10 bit for SDR" is disabled (default when is an SDR display).~

none

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererDXVA.cpp [hunk 1]
```diff
@@ -144,9 +144,9 @@ bool CRendererDXVA::Configure(const VideoPicture& picture, float fps, unsigned o
           m_enumerator->SupportedConversions(m_conversionsArgs);
       if (!conversions.empty())
       {
-        const ProcessorConversion chosenConversion = ChooseConversion(
-            conversions, picture.colorBits,
-            static_cast<AVColorTransferCharacteristic>(picture.color_transfer), m_tryVSR);
+        const ProcessorConversion chosenConversion =
+            ChooseConversion(conversions, picture.colorBits,
+                             static_cast<AVColorTransferCharacteristic>(picture.color_transfer));
```
Comment (CrystalP) [inline xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererDXVA.cpp:149]: The member variable was created with the idea of reacting to settings changes during playback with CheckVideoParameters(), but I stopped half-way in the PR, and the value is calculated only once in Configure. I should have removed the member, which is what I suggest you instead.

**Comment by thexai** [comment]
Oh really? Are you saying that the lack of quality of an imaginary encode is a reason not to merge this PR?

Have you thought that 10 bits is used anyway for 8 bit and 10 bit encodes when "Use 10 bits for SDR" setting is enabled?

**Comment by CrystalP** [comment]
I think it is a reason to pause. Why the change of the 3rd commit? Pixel shaders would use a float texture with extra cost.

My comment was not a rejection, but means "I'm not sure, let's get more information".
I don't know if those 2 extra bits contain valuable information (maybe, even if they didn't come from the source), or if they are like noise. Do you know?
If neither of us knows, asking other team members seems reasonable to me.

Same question applies to "Use 10 bits for SDR" yes, but it also has the justification of matching the texture format of the screen.

File: xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererDXVA.cpp (hunk 1 above)
Comment (CrystalP) [reply xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererDXVA.cpp:149]: The commit is fine. The member is already used in CheckVideoParameters, the editor didn't highlight it earlier and I missed it.

**Comment by thexai** [comment]
I still think that 10 bit intermediate target is "better" for 10 bit SDR but I leave this change for later because it was not the initial intention of this PR and probably some change in the documentation / help texts should be made too.

**Comment by CrystalP** [comment]
I thought about it, looked for information, and I'll agree with something a bit different and yes other PR would be good, with inclusion of Pixel Shaders for consistency.

From what I found, 10bit IT would be good even for 8 bit source / 8 bit encoding because the conversion from YCbCr to RGB is not exact in 8 bit, you need 10 bits to be able to round-trip.
The setting is called "High precision" and an accurate conversion makes sense.

By that logic, 10 bit YCbCr to 10 bit RGB is also not perfect, but the difference is so small it's imperceptible (also dxva processors like Intel  only offer up to RGB10 output, there is no float format output)

File: xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererDXVA.cpp [hunk 2]
```diff
@@ -498,13 +498,9 @@ ProcessorConversion CRendererDXVA::ChooseConversion(
...
-    tryHQ = (DX::Windowing()->GetBackBuffer().GetFormat() == DXGI_FORMAT_R10G10B10A2_UNORM);
-
-    if (DX::Windowing()->IsHighPrecisionProcessingSettingEnabled() && sourceBits > 8 &&
-        (colorTransfer == AVCOL_TRC_SMPTE2084 || colorTransfer == AVCOL_TRC_ARIB_STD_B67))
-    {
-      tryHQ = true;
-    }
+    tryHQ = (DX::Windowing()->GetBackBuffer().GetFormat() == DXGI_FORMAT_R10G10B10A2_UNORM) ||
+            (DX::Windowing()->IsHighPrecisionProcessingSettingEnabled() && sourceBits > 8 &&
+             (colorTransfer == AVCOL_TRC_SMPTE2084 || colorTransfer == AVCOL_TRC_ARIB_STD_B67));
```
Comment (CrystalP) [inline xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererDXVA.cpp:503]: if you're going to submit another change, let's leave this alone for now?

**Review by CrystalP — APPROVED** [review]
Removing clutter is a good thing!

**Pull request merged**